superset/assets/javascripts/dashboard/reducers/allSlices.js

```diff
@@ -0,0 +1,74 @@
+import * as actions from '../actions/allSlices';
+import { t } from '../../locales';
+
+export const initAllSlices = {
+  slices: {},
+  isLoading: true,
+  errorMsg: null,
+  lastUpdated: 0,
+};
+
+export default function allSlicesReducer(state = initAllSlices, action) {
+  const actionHandlers = {
+    [actions.UPDATE_SLICE_NAME]() {
+      const updatedSlice = {
+        ...state.slices[action.key],
+        slice_name: action.sliceName,
+      };
+      const updatedSlices = {
+        ...state.slices,
+        [action.key]: updatedSlice,
+      };
+      return { ...state, slices: updatedSlices };
+    },
+    [actions.FETCH_ALL_SLICES_STARTED]() {
+      return {
+        ...state,
+        isLoading: true,
+      }
+    },
+    [actions.SET_ALL_SLICES]() {
+      let slices = { ...state.slices };
+      action.result.map(slice => (
+        slices['slice_' + slice.id] = {
```

Comment:
why not just use `slice.id`? then you have to encode / decode the same way everywhere and it doesn't match `slice.id`